mock_ros2_project/service_server.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+"""
+ROS2 Service Server Example
+
+This node demonstrates how to create a service server that provides
+an addition service using the AddTwoInts service type.
+"""
+
+import rclpy
+from rclpy.node import Node
+from example_interfaces.srv import AddTwoInts
+
+
+class ServiceServer(Node):
+    """Service server node that provides addition service.
+    
+    This node creates a service server that accepts two integers
+    and returns their sum using the AddTwoInts service interface.
+    """
+    
+    def __init__(self) -> None:
+        """Initialize the service server node."""
+        super().__init__('service_server')
+        
+        # Create service for adding two integers
+        self.srv = self.create_service(
```

Comment:
You may want to pass an explicit `qos_profile` argument (e.g., `rclpy.qos.qos_profile_services_default`) to `create_service()` so QoS behaviour is explicit and not left to middleware defaults.

```suggestion
        self.srv = self.create_service(
            AddTwoInts,
            'add_two_ints',
            self.add_two_ints_callback,
            qos_profile=rclpy.qos.qos_profile_services_default
        )
```